addons/hr/models/hr_employee.py

```diff
@@ -219,10 +219,10 @@ def _cron_check_work_permit_validity(self):
                     user_id=responsible_user_id)
         employees_scheduled.write({'work_permit_scheduled_activity': True})
 
-    def read(self, fields, load='_classic_read'):
+    def read(self, fields=None, load='_classic_read'):
         if self.check_access_rights('read', raise_exception=False):
             return super(HrEmployeePrivate, self).read(fields, load=load)
-        private_fields = set(fields).difference(self.env['hr.employee.public']._fields.keys())
+        private_fields = set(fields or []).difference(self.env['hr.employee.public']._fields.keys())
```

Comment:
`fields=None` means all fields, so I suspect it should be
```suggestion
        private_fields = set(fields or self._fields).difference(self.env['hr.employee.public']._fields)
```